src/main/java/seedu/address/model/animal/exceptions/AnimalNotFoundException.java

```diff
@@ -0,0 +1,8 @@
+package seedu.address.model.animal.exceptions;
+
+/**
+ * Signals that the operation is unable to find the specified animal.
+ */
+public class AnimalNotFoundException extends RuntimeException {}
```

Comment:
Should these custom exceptions be Checked Exceptions instead of Unchecked Exceptions? Reason being that we already anticipated the possibility of the user entering an invalid Animal to look for.

Reply:
Another viewpoint would be that the input of methods throwing these exception should already be validated/sanitized i.e. responsibility of validating inputs is in the caller. Then it would be fine to keep it as-is, but I would suggest including `@throws AnimalNotFoundException` in the javadocs of the relevant methods